src/server/pegasus_write_service.cpp

```diff
@@ -296,15 +307,23 @@ void pegasus_write_service::set_default_ttl(uint32_t ttl) { _impl->set_default_t
 
 void pegasus_write_service::clear_up_batch_states()
 {
-    uint64_t latency = dsn_now_ns() - _batch_start_time;
-    for (dsn::perf_counter *pfc : _batch_qps_perfcounters)
-        pfc->increment();
-    for (dsn::perf_counter *pfc : _batch_latency_perfcounters)
-        pfc->set(latency);
-
-    _batch_qps_perfcounters.clear();
-    _batch_latency_perfcounters.clear();
+#define PROCESS_WRITE_BATCH(op)                                                                    \
+    do {                                                                                           \
+        _##op##_counter->increment_by(static_cast<int64_t>(_##op##_batch_size));                   \
+        for (uint32_t i = 0; i < _##op##_batch_size; ++i) {                                        \
+            _##op##_latency_ns->set(latency_ns);                                                   \
```

Comment:
How about encapsulate another function like `set(count, value)` for percentile metrics?

Reply:
Good idea, I'll encapsulate this interface which is actually needed by unit-tests.